app/src/server/api/routers/item.ts

```diff
@@ -287,7 +290,28 @@ export const itemRouter = createTRPCRouter({
     .output(baseServerResponse)
     .mutation(async ({ ctx, input }) => {
       // Fetch
-      const useritems = await fetchUserItems(ctx.drizzle, ctx.userId);
+      const [useritems, user] = await Promise.all([
+        fetchUserItems(ctx.drizzle, ctx.userId),
+        fetchUser(ctx.drizzle, ctx.userId),
+      ]);
+      const useritem = useritems.find((i) => i.id === input.userItemId);
+      // Definitions & Guard
+      if (!useritem) return errorResponse("User item not found");
+      if (useritem.storedAtHome) return errorResponse("Fetch at home first");
+      if (user.level < useritem.item.requiredLevel) {
+        return errorResponse(`You need to be level ${useritem.item.requiredLevel} to equip this item`);
+      }
+      const doEquip = !useritem.equipped || useritem.equipped !== input.slot;
+      const info = useritem.item;
```

Comment:
Think we should move this logic into the toggleEquipItem function, so that it also applies to anywhere else where we use the toggleEquipItem function. Also the branch seems to need a rebase on main